packages/mdc-ripple/adapter.ts

```diff
@@ -42,75 +40,54 @@
  * for more details.
  * https://github.com/material-components/material-components-web/blob/master/docs/code/architecture.md
  *
- * @record
  */
-class MDCRippleAdapter {
-  /** @return {boolean} */
-  browserSupportsCssVars() {}
-
-  /** @return {boolean} */
-  isUnbounded() {}
-
-  /** @return {boolean} */
-  isSurfaceActive() {}
-
-  /** @return {boolean} */
-  isSurfaceDisabled() {}
-
-  /** @param {string} className */
-  addClass(className) {}
-
-  /** @param {string} className */
-  removeClass(className) {}
-
-  /** @param {!EventTarget} target */
-  containsEventTarget(target) {}
-
-  /**
-   * @param {string} evtType
-   * @param {!Function} handler
-   */
-  registerInteractionHandler(evtType, handler) {}
-
-  /**
-   * @param {string} evtType
-   * @param {!Function} handler
-   */
-  deregisterInteractionHandler(evtType, handler) {}
-
-  /**
-   * @param {string} evtType
-   * @param {!Function} handler
-   */
-  registerDocumentInteractionHandler(evtType, handler) {}
-
-  /**
-   * @param {string} evtType
-   * @param {!Function} handler
-   */
-  deregisterDocumentInteractionHandler(evtType, handler) {}
-
-  /**
-   * @param {!Function} handler
-   */
-  registerResizeHandler(handler) {}
-
-  /**
-   * @param {!Function} handler
-   */
-  deregisterResizeHandler(handler) {}
-
-  /**
-   * @param {string} varName
-   * @param {?number|string} value
-   */
-  updateCssVariable(varName, value) {}
-
-  /** @return {!ClientRect} */
-  computeBoundingRect() {}
-
-  /** @return {{x: number, y: number}} */
-  getWindowPageOffset() {}
+
+interface Point {
+  x: number;
+  y: number;
+}
+
+interface MDCRippleAdapter {
+  browserSupportsCssVars(): boolean;
+
+  isUnbounded(): boolean;
+
+  isSurfaceActive(): boolean;
+
+  isSurfaceDisabled(): boolean;
+
+  addClass(className: string): void;
+
+  removeClass(className: string): void;
+
+  containsEventTarget(target: EventTarget | null): boolean;
+
+  registerInteractionHandler<K extends keyof GlobalEventHandlersEventMap>(
```

Comment:
ooh fancy! I didn't know this was a thing :)

I'd suggest putting some aliases in a common place somewhere to make the types more readable:

```ts
type EventType = keyof GlobalEventHandlersEventMap;
type SpecificEventListener<K extends EventType> = (evt: GlobalEventHandlersEventMap[K]) => void;
```
Then these would look like:

```ts
registerInteractionHandler<E extends EventType>(evtType: E, handler: SpecificEventListener<E>): void;
registerResizeHandler(handler: SpecificEventListener<'resize'>): void;
```

Reply:
yaa I like that :+1: